src/main/java/br/com/dandrade/viagens/controllers/dto/output/FlightDto.java

```diff
@@ -0,0 +1,66 @@
+package br.com.dandrade.viagens.controllers.dto.output;
+
+import br.com.dandrade.viagens.models.Stretch;
+import br.com.dandrade.viagens.models.Ticket;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FlightDto {
+    private final String company;
+    private final String destiny;
+    private final String origin;
+    private final List<StretchDto> stretchs;
+    private final List<String> connectionsOrScale;
+    private final Long totalTimeOfScale;
+
+    public FlightDto( Ticket ticket ) {
+        this.company = ticket.getCompanyName();
+        this.destiny = ticket.getDestiny();
+        this.origin = ticket.getOrigin();
+        this.stretchs = StretchDto.create(ticket.getStretchs(), ticket.getDeparture());
```

Comment:
O método tem 2 argumentos e as duas fontes são do ticket. Me pergunto pq não passamos o Ticket inteiro. Se for algo relacionado a não querer dar acesso a mais coisa do Ticket do que precisamos, você pode criar uma interface, fazer o Ticket implementar e referenciar a interface no método. Manteria claro a necessidade do método. Esse é o princípio do Solid que mais acho útil, a tal da interface segregation.

Reply:
Eu realmente queria evitar dar acesso a mais informação que o necessário. Acho "interface segregation" uma boa, só tenho um pouco de dificuldade em dar um nome a "interface" que tenha uma boa semântica.

Reply:
Realmente a definição do nome não é fácil.. Por outro lado, um simples atalho da ide te fala quais são as implementações. Isso se você também não tiver colocado um javadoc linkando para uma implementação. Eu colocaria algo como StretchMinimalInformation, StretchRequiredInformation, etc. Geralmente o nome delas vai flertar com algo que lembre parâmetros...

Reply:
> Muito legal essa feature. Muitas lógicas apareceram e aí fica bem mais desafiador. Inclusive para revisar!

Sim :)

Reply:
Fiz, mas na hora fiquei um pouco indeciso sobre se fazia o ticket implementar a ‘interface’ ou retornava uma classe anonima delegando as chamadas dos métodos... optei por implementar.